### PR TITLE
Multipart mime 3487 v38

### PR DESCRIPTION
**Pull request by catenacyber**
Link to [redmine](https://redmine.openinfosecfoundation.org/projects/suricata/issues) ticket:
https://redmine.openinfosecfoundation.org/issues/3487

Describe changes:
- convert HTTP to use new rust mime parser
- convert SMTP to use new rust mime parser
- json schema : add email.received array

Follows #11115 with rebase to get more SV test + adding a field in json schema from this test

SV_BRANCH=https://github.com/OISF/suricata-verify/pull/1851

### CODE REVIEW

File: rust/src/mime/mod.rs [hunk 1]
```diff
@@ -1,4 +1,4 @@
-/* Copyright (C) 2021 Open Information Security Foundation
+/* Copyright (C) 2022 Open Information Security Foundation
```
Comment (jlucovsky) [inline rust/src/mime/mod.rs:1]: nit: date

File: rust/src/mime/smtp.rs [hunk 2]
```diff
@@ -0,0 +1,844 @@
+/* Copyright (C) 2022 Open Information Security Foundation
```
Comment (jlucovsky) [inline rust/src/mime/smtp.rs:1]: date, here and everywhere

File: src/app-layer-smtp.c [hunk 3]
```diff
@@ -314,75 +309,48 @@ static void SMTPConfigure(void) {
         if (extract_urls_schemes) {
             ConfNode *scheme = NULL;
 
+            SCMimeSmtpConfigExtractUrlsSchemeReset();
             TAILQ_FOREACH (scheme, &extract_urls_schemes->head, next) {
-                /* new_val_len: scheme value from config e.g. 'http' + '://' + null terminator */
-                size_t new_val_len = strlen(scheme->val) + 3 + 1;
-                if (new_val_len > UINT16_MAX) {
+                size_t scheme_len = strlen(scheme->val);
+                if (scheme_len > UINT16_MAX - SCHEME_SUFFIX_LEN) {
```
Comment (jlucovsky) [inline src/app-layer-smtp.c:315]: nit: even though `-` has higher precedence than `>`, I'd prefer paren's around `UINT16_MAX - SCHEME_SUFFIX_LEN)`

**Review by jlucovsky — COMMENTED** [review]
just a few nits -- copyright dates.

LGTM

**Comment by suricata-qa** [comment]
Information: QA ran without warnings. 

Pipeline 20760

**Review by victorjulien — APPROVED** [review]

File: src/app-layer-smtp.c (hunk 3 above)
Comment (catenacyber) [reply src/app-layer-smtp.c:315]: Why so ?

**Comment by catenacyber** [comment]
> Needs rebase due to a conflict.

Rebased in https://github.com/OISF/suricata/pull/11157

> Can you update the years while at it?

Confused about this :
Jason told The year doesn't hold much value - but if a PR was first submitted in 2022 but not merged until 2024, it should keep its 2022 copyright date.
I still updated some years

**Pull request closed without merging**